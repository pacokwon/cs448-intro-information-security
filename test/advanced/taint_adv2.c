#include "homework.h"

int main() {
	int x = source(); 
	int result;

	if (x > 0){
		if (x > 1){
			result = 0;
			for (int i=1; i<=x; i++){
				result = result * i;
			}
			sink(result);
		}
		else if (x == 1){
			result = 1;
			sink(result);
		}
		else {
			result = sanitizer(x);
			sink(result);
		}
	}
	
	sink(x);
	return 0;
}
