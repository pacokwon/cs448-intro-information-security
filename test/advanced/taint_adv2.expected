Potential Tainted-flow @ advanced/taint_adv2.c:main:25:2 (call void @sink(i32 %call), !dbg !50)
